src/interfaces/chain.cpp

```diff
@@ -160,8 +160,8 @@ class NotificationsHandlerImpl : public Handler, CValidationInterface
     void disconnect() override
     {
         if (m_notifications) {
-            m_notifications = nullptr;
```

Comment:
This change seems clearly correct. Good catch! Maybe `m_notifications` should be an atomic pointer too since `m_notifications` is get and set from different threads, though probably the synchronization done by boost signals & the scheduler prevents bugs from it not being atomic

Reply:
> Maybe m_notifications should be an atomic pointer too since m_notifications is get and set from different threads, though probably the synchronization done by boost signals & the scheduler prevents bugs from it not being atomic

You're right